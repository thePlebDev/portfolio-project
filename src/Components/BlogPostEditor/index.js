import React,{useState} from 'react';
import styled from 'styled-components';

import Form from '../Form'
import AdminTextInput from '../AdminTextInput';
import SubmitButton from '../SubmitButton'
import TextArea from '../TextArea'

import useBlogPost from '../../Hooks/useBlogPost'
import validation from '../../utils/Validation'

const Container = styled.div`

  width:80%;
  text-align:center;
  margin:0 auto;
`

const BlogPostEditor =({useApiCal=useBlogPost})=>{
  const {state,errors,handleChange,handleSubmit} = useApiCal(validation.blogPost)

  return(
    <Container data-testid="container">
        <Form onSubmit={handleSubmit}>
          <AdminTextInput data-testid="input" state={state.imageLink} name={'imageLink'} handleChange={handleChange} error={errors.imageLink} />
          <AdminTextInput data-testid="input" state={state.title} name={'title'} handleChange={handleChange} error={errors.title} />
          <AdminTextInput data-testid="input" state={state.description} name={'description'} handleChange={handleChange} error={errors.description} />
          <AdminTextInput data-testid="input" state={state.filters} name={'filters'} handleChange={handleChange} error={errors.filters} />
          <TextArea data-testid="input" value={state.body} handleChange={handleChange} name={'body'} />
          <SubmitButton data-testid="button">Click</SubmitButton>
        </Form>
    </Container>
  )
}

export default BlogPostEditor
