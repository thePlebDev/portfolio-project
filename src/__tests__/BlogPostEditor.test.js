import React from 'react';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import BlogPostEditor from '../Components/BlogPostEditor';

//1) get a snapshot
//2) test the props
//3) make sure the major parts render
//4)test the actions

describe('testing the BlogPostEditor Component',()=>{
  const useMockFunc = jest.fn(()=>{
    return{
      state:'state',
      errors:'errors',
      handleChange:function(){},
      handleSubmit:function(){}
    }
  })
  let wrapper;
  beforeEach(()=>{
    wrapper = shallow(<BlogPostEditor useApiCal={useMockFunc}/>)
  })
  it('should test the snapshot',()=>{
    const tree = renderer.create(<BlogPostEditor />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should make sure that everything renders properly',()=>{
    expect(wrapper.find('[data-testid="container"]').length).toBe(1)
    expect(wrapper.find('[data-testid="input"]').length).toBe(5)
    expect(wrapper.find('[data-testid="button"]').length).toBe(1)
  })
  it('should use the mock function 3 times',()=>{
    expect(useMockFunc.mock.calls.length).toBe(3)
  })
})
