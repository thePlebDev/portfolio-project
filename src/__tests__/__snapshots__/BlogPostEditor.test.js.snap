// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the BlogPostEditor Component should test the snapshot 1`] = `
<div
  className="sc-jSFkmK hDoHkc"
  data-testid="container"
>
  <form
    className="sc-bdnylx kFpyQD"
    onSubmit={[Function]}
  >
    <div
      className="sc-gtssRu iMKyf"
      data-testid="container"
    >
      <label
        for="username"
      >
        <input
          className="sc-dlnjPT dFiXol"
          data-testid="input"
          id="username"
          name="imageLink"
          onChange={[Function]}
          placeholder="imageLink"
          type="text"
          value=""
        />
      </label>
    </div>
    <div
      className="sc-gtssRu iMKyf"
      data-testid="container"
    >
      <label
        for="username"
      >
        <input
          className="sc-dlnjPT dFiXol"
          data-testid="input"
          id="username"
          name="title"
          onChange={[Function]}
          placeholder="title"
          type="text"
          value=""
        />
      </label>
    </div>
    <div
      className="sc-gtssRu iMKyf"
      data-testid="container"
    >
      <label
        for="username"
      >
        <input
          className="sc-dlnjPT dFiXol"
          data-testid="input"
          id="username"
          name="description"
          onChange={[Function]}
          placeholder="description"
          type="text"
          value=""
        />
      </label>
    </div>
    <div
      className="sc-gtssRu iMKyf"
      data-testid="container"
    >
      <label
        for="username"
      >
        <input
          className="sc-dlnjPT dFiXol"
          data-testid="input"
          id="username"
          name="filters"
          onChange={[Function]}
          placeholder="filters"
          type="text"
          value=""
        />
      </label>
    </div>
    <div>
      <label>
        <textarea
          className="sc-eCApGN bxyVFY"
          name="body"
          onChange={[Function]}
        />
      </label>
    </div>
    <button
      className="sc-hKFyIo derYWF"
      type="submit"
    >
      SEND
    </button>
  </form>
</div>
`;
